validator/services/edgeware_sender.js

```diff
@@ -0,0 +1,72 @@
+const sha3 = require("sha3");
+const polkaTypes = require("@polkadot/types");
+
+/**
+ * Class for intreracting with Bridge smart-contract
+ *
+ * @class EdgewareSender
+ */
+class EdgewareSender {
+  constructor(_bridgeContract) {
+    this.bridgeContract = _bridgeContract;
+  }
+
+  /**
+   * send Harmony data for saving in Bridge deployed in Edgeware
+   *
+   * @param {*} swapHarmMessage - Harmony message for saving in Bridge
+   * @param {*} validator - address of validator
+   * @returns Array of [Uint8Array]
+   */
+  async sendHarmDataToEdgeware(swapHarmMessage, validator) {
+    const tx = await this.bridgeContract.tx.requestSwap(0, -1, swapHarmMessage);
+
+    const result = tx.signAndSend(validator);
```

Comment:
Need to catch block hash here to be able to send it to worker so it can confirm finality.
signAndSend takes a callback as a second parameter, which will be called with two arguments (status and events). If status.isInBlock is true then block hash can be obtained with asInBlock

Reply:
It fixed in commit https://github.com/everstake/harmony_bridge_backend/commit/fe9f5607123252c53d288aa3935d68ad8baeba3d